Storage/SQL/SQLiteBookmarksHelpers.swift

```diff
@@ -22,8 +22,39 @@ public func titleForSpecialGUID(_ guid: GUID) -> String? {
     }
 }
 
+class BookmarkURLTooLargeError: MaybeErrorType {
+    init() {
+    }
+    var description: String {
+        return "URL too long to bookmark."
+    }
+}
+
+extension String {
+    public func truncateToUTF8ByteCount(_ keep: Int) -> String {
```

Comment:
Should this go in Shared/Extensions/StringExtensions.swift instead?

Reply:
I was waiting for you to say that :)

I put it here 'cos it'll keep the size of `Shared` smaller. We can move it if we ever want to use this more generally… at which point we'll likely want to revisit its behavior. Right now it's cheap but aggressive.